service/s3/endpoint_test.go

```diff
@@ -265,36 +265,31 @@ func TestEndpoint(t *testing.T) {
 			expectedSigningName:   "s3-outposts",
 			expectedSigningRegion: "us-gov-east-1",
 		},
-		"Outpost AccessPoint Fips region": {
+		"Outpost AccessPoint FIPS client region": {
 			bucket: "arn:aws-us-gov:s3-outposts:us-gov-east-1:123456789012:outpost:op-01234567890123456:accesspoint:myaccesspoint",
 			config: &aws.Config{
 				EndpointResolver: endpoints.AwsUsGovPartition(),
 				Region:           aws.String("fips-us-gov-east-1"),
 			},
 			expectedErr: "ConfigurationError: client region does not match provided ARN region",
 		},
-		"Outpost AccessPoint Fips region in Arn": {
-			bucket: "arn:aws-us-gov:s3-outposts:fips-us-gov-east-1:123456789012:outpost:op-01234567890123456:accesspoint:myaccesspoint",
+		"Outpost AccessPoint FIPS client region with matching ARN region": {
+			bucket: "arn:aws-us-gov:s3-outposts:us-gov-east-1:123456789012:outpost:op-01234567890123456:accesspoint:myaccesspoint",
 			config: &aws.Config{
-				EndpointResolver:        endpoints.AwsUsGovPartition(),
-				EnforceShouldRetryCheck: nil,
-				Region:                  aws.String("fips-us-gov-east-1"),
-				DisableSSL:              nil,
-				HTTPClient:              nil,
-				S3UseARNRegion:          aws.Bool(true),
+				EndpointResolver: endpoints.AwsUsGovPartition(),
+				Region:           aws.String("fips-us-gov-east-1"),
+				S3UseARNRegion:   aws.Bool(true),
```

Comment:
Since request or client configured with fips region would never work when an ARN is used. Can we change the error thrown to indicate so? 

We can say `Use of ARN is not supported when client or request is configured for FIPs region`